src/lib/common/sol-certificate-impl-linux.c

```diff
@@ -92,19 +92,20 @@ sol_cert_load_from_file(const char *filename)
 
     SOL_NULL_CHECK(filename, NULL);
 
-    SOL_PTR_VECTOR_FOREACH_IDX (&storage, cert, idx) {
-        if (streq(cert->filename, filename)) {
-            cert->refcnt++;
-            return cert;
-        }
-    }
-
     path = find_cert(filename, search_paths);
     if (path == NULL) {
         SOL_WRN("Certificate not found: %s", filename);
         return NULL;
     }
 
+    SOL_PTR_VECTOR_FOREACH_IDX (&storage, cert, idx) {
+        if (streq(cert->filename, path)) {
```

Comment:
I don't get this order change, the commit doesn't clarify much either. The idea, AFAIU, is to only load once, so first you check in the storage, then if not found, you load it.